plugins/cccongress.py

```diff
@@ -0,0 +1,275 @@
+@asyncio.coroutine
+def cccongress_update_cron(bot):
+    """Update schedule in December"""
+
+    yield from _update_cache(bot)
+
+
+@cron('* * 27-30 12 *')
+@asyncio.coroutine
+def cccongress_announce_next_talks(bot):
+    """Announce next talks"""
+
+    config = BYTEBOT_PLUGIN_CONFIG['cccongress']
+
+    announcelist = []
+    for hall in _get_halls():
+        """Get next event for hall"""
+        event = _get_talk(hall, 1)
+        if event is not None:
+            """Check if event starts in a few minutes"""
+            event_start = dateutil.parser.parse(event['date'])
+            designated_start = \
+                datetime.datetime.now(pytz.timezone('Europe/Berlin')) + \
+                datetime.timedelta(minutes=config['announce_minutes'])
+            if event_start == designated_start:
+                """Save hall informations in event and add to announcelist"""
+                event['hall'] = hall
+                announcelist.append(event)
+
+    """Announce found events"""
+    if len(announcelist) > 0:
+        bot.privmsg(
+            bot.config.autojoins[0],
+            "The following talks will start in %d minutes" %
+            config['announce_minutes']
+        )
+        for event in announcelist:
+            yield from _output_single_talk(event['hall'],
+                                           event,
+                                           bot,
+                                           bot.config.autojoins[0])
+
+
+@asyncio.coroutine
+def _output_talks(slot, bot, target):
+    """Output talks
+
+        slot: Output talks running now (slot = 0) or next (slot = 1)
+    """
+
+    eventcounter = 0
+    for hall in _get_halls():
+        event = _get_talk(hall, slot)
+        if event is not None:
+            eventcounter += 1
+            yield from _output_single_talk(hall, event, bot, target)
+
+    if eventcounter == 0:
+        bot.privmsg(target, "No talks found.")
+
+
+@asyncio.coroutine
+def _output_single_talk(hall, event, bot, target):
+    """Output single talk
+
+        hall: Name of hall
+        event: Event informations
+    """
+
+    bot.privmsg(target,
+                "[%s] %s %s" % (hall, event['start'], event['title']))
+    bot.privmsg(
+        target,
+        "%s (%s / %s)" %
+        ('{:<14}'.format(''), event['language'], _get_persons(event))
+    )
+
+
+@asyncio.coroutine
+def _scheduleinformations(bot, target):
+    """Output informations about schedule"""
+
+    config = BYTEBOT_PLUGIN_CONFIG['cccongress']
+    json_data = _get_json_data()
+
+    bot.privmsg(
+        target,
+        "%s (%s)" %
+        (json_data['schedule']['conference']['title'],
+         json_data['schedule']['conference']['acronym'])
+    )
+
+    file_mtime = os.path.getmtime(config['cache'])
+    file_dtime = datetime.datetime.fromtimestamp(file_mtime)
+    bot.privmsg(
+        target,
+        "Schedule was last updated on %s" %
+        file_dtime.strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
This line looks dangerous without more indent, because you can't distinguish the logical units here.